src/server/pipeline.py

```diff
@@ -317,12 +353,88 @@ async def handle_message(
             yield connection, outbound_message, global_connections
             return
         elif full_message.function == "pong":
+            if connection.connection_type == NodeType.FULL_NODE:
+                await global_connections.update_connection_time(
+                    connection.get_peer_info()
+                )
             return
 
-        f_with_peer_name = getattr(api, full_message.function + "_with_peer_name", None)
+        # Peer gossip message handle.
+        if full_message.function == "request_peers":
+            # Handle here only full nodes peer gossip, and let
+            # the introducer use its own function.
+            if global_connections.local_type == NodeType.FULL_NODE:
+                if global_connections is None:
+                    return
+                # Prevent a fingerprint attack.
+                if connection.is_outbound:
```

Comment:
Why are we not responding with peers to outbound connections?

Reply:
I copied this behaviour from bitcoin code:

if (strCommand == NetMsgType::GETADDR) {
// This asymmetric behavior for inbound and outbound connections was introduced
// to prevent a fingerprinting attack: an attacker can send specific fake addresses
// to users' AddrMan and later request them by sending getaddr messages.
// Making nodes which are behind NAT and can only make outgoing connections ignore
// the getaddr message mitigates the attack.
if (!pfrom->fInbound)
return true;
}